VultisigApp/VultisigApp/Views/Referral/ReferralTransactionDetailsView.swift

```diff
@@ -108,27 +145,37 @@ struct ReferralTransactionDetailsView: View {
             Spacer()
             
             if let icon {
-                Circle()
-                    .foregroundColor(.black)
+                Image(icon)
+                    .resizable()
                     .frame(width: 16, height: 16)
+                    .cornerRadius(32)
             }
             
             Text(description)
                 .foregroundColor(.neutral0)
             
             if let bracketValue {
-                Text("(\(title))")
+                Text("(\(bracketValue))")
                     .foregroundColor(.extraLightGray)
             }
         }
         .font(.body14BrockmannMedium)
+        .foregroundColor(.neutral0)
     }
     
     var button: some View {
+        NavigationLink {
```

Comment:
based on my previous experience , this might cause the app to freeze , but I am not sure , can you test it out?